src/Symfony/Component/Encryption/Ciphertext.php

```diff
@@ -0,0 +1,183 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Encryption;
+
+use Symfony\Component\Encryption\Exception\DecryptionException;
+use Symfony\Component\Encryption\Exception\MalformedCipherException;
+
+/**
+ * This class is responsible over the payload API.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ *
+ * @internal
+ */
+class Ciphertext
+{
+    /**
+     * @var string algorithm for the encryption
+     */
+    private $algorithm;
+
+    /**
+     * @var string
+     */
+    private $version;
+
+    /**
+     * @var string the encoded payload
+     */
+    private $payload;
+
+    /**
+     * @var string nonce for the algorithm
+     */
+    private $nonce;
+
+    /**
+     * @var array<string, string> additional headers
+     */
+    private $headers = [];
+
+    private function __construct()
+    {
+    }
+
+    /**
+     * @param array<string, string> $headers with ascii keys and values
+     */
+    public static function create(string $algorithm, string $ciphertext, string $nonce, array $headers = []): self
+    {
+        $model = new self();
+        $model->algorithm = $algorithm;
+        $model->payload = $ciphertext;
+        $model->nonce = $nonce;
+        $model->headers = $headers;
```

Comment:
The Nonce is used for most (not all) encryption. We could move it into the headers instead. That would mean that `headers` amore more like `context`.

Reply:
sounds good to me

Reply:
I started to remove the nonce, but I change my mind. 

The headers must be ascii characters because we are json encoding it. 
However, good nonces do not restrict themselves to ascii.